Problems trying to compile syslog-ng 3.6.4 on Solaris 11.2 (SPARC)

From OSE mail list:
```
Have compiled and installed eventlog-0.2.12  (into /usr/local/syslog-ng).
 
Now trying to compile syslog-ng I’m hitting the error shown at the end.
 
Following commands were used to configure the environment for the compile
 
export PKG_CONFIG_PATH=/usr/local/syslog-ng/lib/pkgconfig
./configure --prefix=/usr/local/syslog-ng --enable-sun-streams --enable-tcp-wrapper --enable-ipv6 --enable-sql --enable-ssh --with-ld-library-path=/usr/local/syslog-ng/lib --disable-amqp --enable-sun-door
 
Any suggestions ?
 
Thanks

Ian
 
  CC     modules/afsocket/modules_afsocket_libafsocket_notls_la-transport-mapper-unix.lo
  CC     modules/afsocket/modules_afsocket_libafsocket_notls_la-transport-unix-socket.lo
modules/afsocket/transport-unix-socket.c: In function â_format_proc_file_nameâ:
modules/afsocket/transport-unix-socket.c:50:3: warning: format â%dâ expects argument of type âintâ, but argument 4 has type âpid_tâ [-Wformat=]
   g_snprintf(buf, buflen, "/proc/%d/%s", pid, proc_file);
   ^
modules/afsocket/transport-unix-socket.c: In function â_unix_socket_readâ:
modules/afsocket/transport-unix-socket.c:219:6: error: âstruct msghdrâ has no member named âmsg_controlâ
   msg.msg_control = ctlbuf;
      ^
modules/afsocket/transport-unix-socket.c:220:6: error: âstruct msghdrâ has no member named âmsg_controllenâ
   msg.msg_controllen = sizeof(ctlbuf);
      ^
modules/afsocket/transport-unix-socket.c: In function âlog_transport_unix_dgram_socket_newâ:
modules/afsocket/transport-unix-socket.c:261:3: warning: implicit declaration of function âsocket_set_pass_credentialsâ [-Wimplicit-function-declaration]
   socket_set_pass_credentials(fd);
   ^
modules/afsocket/transport-unix-socket.c: At top level:
modules/afsocket/transport-unix-socket.c:36:1: warning: â_add_nv_pair_intâ defined but not used [-Wunused-function]
_add_nv_pair_int(LogTransportAuxData *aux, const gchar *name, gint value)
^
modules/afsocket/transport-unix-socket.c:102:1: warning: â_add_nv_pair_proc_read_unless_unsetâ defined but not used [-Wunused-function]
_add_nv_pair_proc_read_unless_unset(LogTransportAuxData *aux, const gchar *name, pid_t pid, const gchar *proc_file, const gchar *unset_value)
^
modules/afsocket/transport-unix-socket.c:115:1: warning: â_add_nv_pair_proc_read_argvâ defined but not used [-Wunused-function]
_add_nv_pair_proc_read_argv(LogTransportAuxData *aux, const gchar *name, pid_t pid, const gchar *proc_file)
^
modules/afsocket/transport-unix-socket.c:135:1: warning: â_add_nv_pair_proc_readlinkâ defined but not used [-Wunused-function]
_add_nv_pair_proc_readlink(LogTransportAuxData *aux, const gchar *name, pid_t pid, const gchar *proc_file)
^
gmake[2]: *** [modules/afsocket/modules_afsocket_libafsocket_notls_la-transport-unix-socket.lo] Error 1
gmake[1]: *** [all-recursive] Error 1
gmake: *** [all] Error 2

The contents of this e-mail are confidential and for the exclusive use of the intended recipient. If you are not the intended recipient you should not read, copy, retransmit or disclose its contents. If you have received this email in error please delete it from your system immediately and notify us either by email or telephone. The views expressed in this communication may not necessarily be the views held by McLaren Technology Group Limited. 
McLaren Technology Group Limited | McLaren Technology Centre | Chertsey Road | Woking | Surrey | GU21 4YH | UK | Company Number: 01967715

```

--- a/configure.ac
+++ b/configure.ac
@@ -495,6 +495,10 @@ AC_CHECK_MEMBER(struct tm.tm_gmtoff,AC_DEFINE(HAVE_STRUCT_TM_TM_GMTOFF,1,[Whethe
 #include <time.h>
 #endif])
 
+AC_CHECK_MEMBER(struct msghdr.msg_control,AC_DEFINE(HAVE_CTRLBUF_IN_MSGHDR,1,[Whether you have msg_control field in msghdr in socket.h]),,[
+#include <sys/socket.h>
+])
+
 AC_CACHE_CHECK(for I_CONSLOG, blb_cv_c_i_conslog,
   [AC_EGREP_CPP(I_CONSLOG,
 [

--- a/modules/afsocket/transport-unix-socket.c
+++ b/modules/afsocket/transport-unix-socket.c
@@ -207,8 +207,14 @@ _unix_socket_read(gint fd, gpointer buf, gsize buflen, LogTransportAuxData *aux)
   gint rc;
   struct msghdr msg;
   struct iovec iov[1];
-  gchar ctlbuf[32];
   struct sockaddr_storage ss;
+#if defined(HAVE_CTRLBUF_IN_MSGHDR)
+  gchar ctlbuf[32];
+  msg.msg_control = *ctlbuf;
+  msg.msg_controllen = sizeof(ctlbuf);
+#endif
+
+
 
   msg.msg_name = (struct sockaddr *) &ss;
   msg.msg_namelen = sizeof(ss);
@@ -216,8 +222,6 @@ _unix_socket_read(gint fd, gpointer buf, gsize buflen, LogTransportAuxData *aux)
   msg.msg_iov = iov;
   iov[0].iov_base = buf;
   iov[0].iov_len = buflen;
-  msg.msg_control = ctlbuf;
-  msg.msg_controllen = sizeof(ctlbuf);
   do
     {
       rc = recvmsg(fd, &msg, 0);

--- a/modules/afsocket/unix-credentials.h
+++ b/modules/afsocket/unix-credentials.h
@@ -30,13 +30,13 @@
 #include "syslog-ng.h"
 
 #if defined(__linux__)
-# if HAVE_STRUCT_UCRED
+# if HAVE_STRUCT_UCRED && HAVE_CTRLBUF_IN_MSGHDR
 # define CRED_PASS_SUPPORTED
 # define cred_t struct ucred
 # define cred_get(c,x) (c->x)
 # endif
 #elif defined(__FreeBSD__)
-# if HAVE_STRUCT_CMSGCRED
+# if HAVE_STRUCT_CMSGCRED && HAVE_CTRLBUF_IN_MSGHDR
 #  define CRED_PASS_SUPPORTED
 #  define SCM_CREDENTIALS SCM_CREDS
 #  define cred_t struct cmsgcred

